StardewArchipelago/ModEntry.cs

```diff
@@ -369,10 +369,10 @@ private void OnDayStarted(object sender, DayStartedEventArgs e)
 
         private void DoBugsCleanup()
         {
-            if (_archipelago.SlotData.Mods.HasMod(ModNames.LUCK) && Game1.player.LuckLevel >= 8 && !_locationChecker.IsLocationChecked("Lucky Lunch Recipe"))
+            if (_archipelago.SlotData.Mods.HasMod(ModNames.LUCK) && Game1.player.LuckLevel >= 8 && !_locationChecker.IsLocationMissingAndExists("Lucky Lunch Recipe"))
             {
                 _locationChecker.AddCheckedLocation("Lucky Lunch Recipe");
-                if (!_archipelago.HasReceivedItem("Lucky Lunch Recipe"))
+                if (_archipelago.SlotData.Chefsanity.HasFlag(Chefsanity.Skills) && !_archipelago.HasReceivedItem("Lucky Lunch Recipe"))

```

Comment:
still need to check if the player actually has the recipe before removing it, no?